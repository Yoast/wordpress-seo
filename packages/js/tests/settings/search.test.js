import Search from "../../src/settings/components/search";
import { shallow } from "enzyme";

jest.mock( "react-router-dom" );

describe( "Search test", () => {
	let wrapper;
	beforeEach( () => {
		wrapper = shallow( <Search /> );
	} );

	it( "Should have 'Quick Search' text on search button", () => {
		expect( wrapper.find( "button" ).first().text() ).toContain( "Quick search..." );
	} );

	it( "should open Modal", () => {
		wrapper.find( "button" ).at( 0 ).simulate( "click" );
		expect( wrapper.find( "Modal" ).prop( "isOpen" ) ).toBe( true );
	} );

	it( "Input should have placeholder", () => {
		wrapper.find( "button" ).at( 0 ).simulate( "click" );
		expect( wrapper.find( "#input-search" ).prop( "placeholder" ) ).toContain( "Search..." );
	} );

	it( "Should not search under 3 characters", () => {
		wrapper.find( "button" ).at( 0 ).simulate( "click" );
		wrapper.find( "#input-search" ).at( 0 ).simulate( "change", { target: { value: "se" } } );
		expect( wrapper.find( "SearchNoResultsContent" ).find( "p" ).text() ).toContain( "Please enter a search term with at least 3 characters." );
	} );

	it( "should match button text to input value", () => {
		wrapper.find( "button" ).at( 0 ).simulate( "click" );
		wrapper.find( "#input-search" ).at( 0 ).simulate( "change", { target: { value: "seo title" } } );
		expect( wrapper.find( "button" ).first().text() ).toContain( "seo title" );
	} );

	it( "Results not found", () => {
		wrapper.find( "button" ).at( 0 ).simulate( "click" );
		wrapper.find( "#input-search" ).at( 0 ).simulate( "change", { target: { value: "random" } } );
		expect( wrapper.find( "SearchNoResultsContent" ).find( "p" ).text() ).toContain( "We couldnâ€™t find anything with that term." );
	} );

	it( "should match search snapshot", () => {
		expect( wrapper.debug() ).toMatchSnapshot();
	} );
} );
