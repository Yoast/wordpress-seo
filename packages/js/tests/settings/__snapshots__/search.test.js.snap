// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search test should match search snapshot 1`] = `
"<Fragment>
  <button id=\\"button-search\\" type=\\"button\\" className=\\"yst-w-full yst-flex yst-items-center yst-bg-white yst-text-sm yst-leading-6 yst-text-slate-500 yst-rounded-md yst-border yst-border-slate-300 yst-shadow-sm yst-py-1.5 yst-pl-2 yst-pr-3 focus:yst-outline-none focus:yst-ring-2 focus:yst-ring-offset-2 focus:yst-ring-primary-500\\" onClick={[Function (anonymous)]}>
    <ForwardRef(SearchIcon) className=\\"yst-flex-none yst-w-5 yst-h-5 yst-mr-3 yst-text-slate-400\\" role=\\"img\\" aria-hidden=\\"true\\" />
    <span className=\\"yst-overflow-hidden yst-whitespace-nowrap yst-text-ellipsis\\">
      Quick search...
    </span>
  </button>
  <Modal id=\\"modal-search\\" onClose={[Function (anonymous)]} isOpen={false} initialFocus={{...}} position=\\"top-center\\" aria-label=\\"Search\\">
    <Modal.Panel hasCloseButton={false}>
      <LiveAnnouncer />
      <ComboboxFn as=\\"div\\" className=\\"yst--m-6\\" onChange={[Function (anonymous)]}>
        <div className=\\"yst-relative\\">
          <ForwardRef(SearchIcon) className=\\"yst-pointer-events-none yst-absolute yst-top-3.5 yst-left-4 yst-h-5 yst-w-5 yst-text-slate-400\\" role=\\"img\\" aria-hidden=\\"true\\" />
          <Input2 id=\\"input-search\\" placeholder=\\"Search...\\" aria-label=\\"Search\\" value=\\"\\" onChange={[Function (anonymous)]} className=\\"yst-h-12 yst-w-full yst-border-0 yst-rounded-lg sm:yst-text-sm yst-bg-transparent yst-px-11 yst-text-slate-800 yst-placeholder-slate-500 focus:yst-outline-none focus:yst-ring-inset focus:yst-ring-2 focus:yst-ring-primary-500 focus:yst-border-primary-500\\" />
          <div className=\\"yst-modal__close\\">
            <button type=\\"button\\" onClick={[Function (anonymous)]} className=\\"yst-modal__close-button\\">
              <span className=\\"yst-sr-only\\">
                Close
              </span>
              <ForwardRef(XIcon) className=\\"yst-h-6 yst-w-6\\" role=\\"img\\" aria-hidden=\\"true\\" />
            </button>
          </div>
        </div>
        <SearchNoResultsContent title=\\"Search\\">
          <p className=\\"yst-text-slate-500\\">
            Please enter a search term with at least 3 characters.
          </p>
        </SearchNoResultsContent>
      </ComboboxFn>
    </Modal.Panel>
  </Modal>
</Fragment>"
`;
